### PR TITLE
Add base earn incentives

### PR DESCRIPTION
**Pull request by drklee3**
Adds incentives and hooks for the Earn module. This essentially just a copy of the swap incentives, but with slight modifications which is mostly naming.

### CODE REVIEW

File: cmd/kava/cmd/root.go [hunk 1]
```diff
@@ -83,8 +83,8 @@ func addSubCmds(rootCmd *cobra.Command, encodingConfig params.EncodingConfig, de
 			genutilcli.InitCmd(app.ModuleBasics, defaultNodeHome),
 		),
 		genutilcli.CollectGenTxsCmd(banktypes.GenesisBalancesIterator{}, defaultNodeHome),
-		migrate.MigrateGenesisCmd(),
-		migrate.AssertInvariantsCmd(encodingConfig),
+		// migrate.MigrateGenesisCmd(),
```
Comment (DracoLi) [inline cmd/kava/cmd/root.go:86]: is this still relevant?

File: proto/kava/incentive/v1beta1/claims.proto [hunk 2]
```diff
@@ -115,3 +115,13 @@ message SavingsClaim {
   repeated MultiRewardIndex reward_indexes = 2
       [(gogoproto.castrepeated) = "MultiRewardIndexes", (gogoproto.nullable) = false];
 }
+
+// SwapClaim stores the swap rewards that can be claimed by owner
```
Comment (DracoLi) [inline proto/kava/incentive/v1beta1/claims.proto:119]:
```suggestion
// EarnClaim stores the earn rewards that can be claimed by owner
```

File: proto/kava/incentive/v1beta1/tx.proto [hunk 3]
```diff
@@ -92,3 +95,15 @@ message MsgClaimSavingsReward {
 
 // MsgClaimSavingsRewardResponse defines the Msg/ClaimSavingsReward response type.
 message MsgClaimSavingsRewardResponse {}
+
+// MsgClaimEarnReward message type used to claim searn rewards
```
Comment (DracoLi) [inline proto/kava/incentive/v1beta1/tx.proto:99]:
```suggestion
// MsgClaimEarnReward message type used to claim earn rewards
```

File: x/earn/types/genesis.go [hunk 4]
```diff
@@ -38,3 +44,30 @@ func DefaultGenesisState() GenesisState {
 		VaultShareRecords{},
 	)
 }
+
+func Kava11GenesisState(accountKeeper AccountKeeper) GenesisState {
```
Comment (DracoLi) [inline x/earn/types/genesis.go:48]: should this be moved to the migration PR?

File: x/incentive/types/msg.go [hunk 5]
```diff
@@ -15,6 +15,7 @@ var (
 	_ sdk.Msg = &MsgClaimDelegatorReward{}
 	_ sdk.Msg = &MsgClaimSwapReward{}
 	_ sdk.Msg = &MsgClaimSavingsReward{}
+	_ sdk.Msg = &MsgClaimEarnReward{}
```
Comment (DracoLi) [inline x/incentive/types/msg.go:18]: probably as a separate PR, but wanted to note that we should also add legacy msg type check here as well.
```
_ legacytx.LegacyMsg = &MsgClaimEarnReward{}
```

File: x/earn/types/genesis.go (hunk 4 above)
Comment (drklee3) [reply x/earn/types/genesis.go:48]: ah right oops, seem to have accidentally included it in a different commit some time ago

File: x/incentive/types/msg.go (hunk 5 above)
Comment (drklee3) [reply x/incentive/types/msg.go:18]: just added to this pr since they all seem to already comply to the interface so it's a small change

**Review by rhuairahrighairidh — APPROVED** [review]
looks good
I added a validation to the genesis for the earn state

**Pull request merged**